docs/docs/ScalaUserGuide/build-bigdl-core.md

```diff
@@ -0,0 +1,125 @@
+## **Build BigDL-core on Different Platforms**
+
+### **Setup Build Environment**
+
+For building BigDL-core, there should have
+
++ JDK 1.7+
++ maven
++ make
++ g++-7
++ Intel Parallel Studio
++ Git.
+
+BigDL-core is a JNI project, `mkl2017-xeon-blas` needs MKL libraries with icc and `bigquant` needs g++-7. We use `maven` + `make` to control the build process where maven for java and make for c/c++ code.
+
+### **Environment on CentOS**
+
+* **Build GCC-7.2**
+
+```bash
+git clone https://github.com/gcc-mirror/gcc.git
```

Comment:
why build gcc from source code?

Reply:
CentOS 7.2 hasn't add gcc-7 to the repo. And it's hard to find one usable. So we should install it from source code.